go-controller/pkg/ovn/controller/services/svc_template_var.go

```diff
@@ -0,0 +1,192 @@
+	for _, nbTemplate := range flattened {
+		if !callback(nbTemplate) {
+			break
+		}
+	}
+}
+
+// getLoadBalancerTemplates returns the TemplateMap of templates referenced
+// by the load balancer. These are pointers to Templates already read from
+// the database which are stored in 'allTemplates'.
+func getLoadBalancerTemplates(lb *nbdb.LoadBalancer, allTemplates TemplateMap) TemplateMap {
+	result := TemplateMap{}
+	for vip, backend := range lb.Vips {
+		for _, s := range []string{vip, backend} {
+			if isTemplateReference(s) {
+				templateName := templateNameFromReference(s)
+				if template, found := allTemplates[templateName]; found {
+					result[template.Name] = template
+				}
+			}
+		}
+	}
+	return result
+}
+
+// listSvcTemplates looks up all chassis template variables. It returns
+// libovsdb.Templates values indexed by template name.
+func listSvcTemplates(nbClient libovsdbclient.Client) (templatesByName TemplateMap, err error) {
+	templatesByName = TemplateMap{}
+
+	templatesList, err := libovsdbops.ListTemplates(nbClient)
+	if err != nil {
+		return
+	}
+
+	for _, nbTemplate := range templatesList {
+		for name, perChassisValue := range nbTemplate.Variables {
+			tv, found := templatesByName[name]
+			if !found {
+				tv = makeTemplate(name)
+				templatesByName[name] = tv
+			}
+			tv.Value[nbTemplate.Chassis] = perChassisValue
+		}
+	}
+	return
+}
+
+func svcCreateTemplateVarOps(nbClient libovsdbclient.Client,
+	ops []libovsdb.Operation, templateVars []TemplateMap) ([]libovsdb.Operation, error) {
+
+	var err error
+
+	forEachNBTemplateInMaps(templateVars, func(nbTemplate *nbdb.ChassisTemplateVar) bool {
+		ops, err = libovsdbops.CreateChassisTemplateVarOps(nbClient, ops, nbTemplate)
+		return err == nil
+	})
+	if err != nil {
+		return nil, err
+	}
+	return ops, nil
+}
+
+func svcCreateOrUpdateTemplateVarOps(nbClient libovsdbclient.Client,
+	ops []libovsdb.Operation, templateVars []TemplateMap) ([]libovsdb.Operation, error) {
+
+	var err error
+
+	forEachNBTemplateInMaps(templateVars, func(nbTemplate *nbdb.ChassisTemplateVar) bool {
+		ops, err = libovsdbops.CreateOrUpdateChassisTemplateVarOps(nbClient, ops, nbTemplate)
+		return err == nil
+	})
+	if err != nil {
+		return nil, err
+	}
+	return ops, nil
+}
+
+func svcDeleteTemplateVarOps(nbClient libovsdbclient.Client,
+	ops []libovsdb.Operation, templateVars []TemplateMap) ([]libovsdb.Operation, error) {
+	var err error
+
+	forEachNBTemplateInMaps(templateVars, func(nbTemplate *nbdb.ChassisTemplateVar) bool {
+		ops, err = libovsdbops.DeleteChassisTemplateVarOps(nbClient, ops, nbTemplate)
+		return err == nil
+	})
+	if err != nil {
+		return nil, err
+	}
+	return ops, nil
+}
+
+func svcCreateOrUpdateTemplateVar(nbClient libovsdbclient.Client, templateVars []TemplateMap) error {
+	ops, err := svcCreateOrUpdateTemplateVarOps(nbClient, nil, templateVars)
+	if err != nil {
+		return err
+	}
+
+	_, err = libovsdbops.TransactAndCheck(nbClient, ops)
+	return err
```

Comment:
This could live in libovsdbops, same mehtod name without the `Ops` in the end.

Reply:
I'm not sure I follow.  It uses the services wrapper.  I left it as is for now.

Reply:
Sorry, I missed it was the wrapper. Ignore.